### PR TITLE
Update pools ping & timeout logic to prevent dropped idle connections

### PR DESCRIPTION
**Pull request by NathanFlurry**

### CODE REVIEW

**Comment by NathanFlurry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/336?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* `main`
  * **#324** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#314** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **#310** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/310?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **#311** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/311?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **#312** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **#313** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/313?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **#315** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **#316** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **#326** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **#325** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **#329** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **#330** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/330?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **#331** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/331?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **#332** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **#333** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/333?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **#334** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/334?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **#335** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/335?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **#336** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/336?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @NathanFlurry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by NathanFlurry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/336?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* `main`
  * **#324** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#314** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **#310** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/310?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **#311** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/311?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **#312** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **#313** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/313?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **#315** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **#316** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **#326** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **#325** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **#329** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **#330** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/330?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **#331** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/331?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **#332** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **#333** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/333?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **#334** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/334?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **#335** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/335?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **#336** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/336?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @NathanFlurry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/infrastructure/TIMEOUTS.md
Comment (AngelOnFira) [inline docs/infrastructure/TIMEOUTS.md:1]: [beautiful](https://app.graphite.dev/macros/rivet-gg)

**Review by AngelOnFira — APPROVED** [review]

File: docs/infrastructure/TIMEOUTS.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+We use long polling (i.e. `watch_index`) to implement real time functionality. This means we need to be cautious about existing timeouts.
+
+Current timeouts:
+
+-   Cloudflare: 100s ([source](https://developers.cloudflare.com/support/troubleshooting/cloudflare-errors/troubleshooting-cloudflare-5xx-errors/#error-524-a-timeout-occurred))
+    -   **Behavior** Returns a 524
+    -   Cannot be configured unless paying for Cloudflare Enterprise
+-   Traefik: 60s ([source](https://github.com/rivet-gg/rivet/blob/c63067ce6e81f97b435e424e576fbd922b14f748/infra/tf/k8s_infra/traefik.tf#L65))
+    -   **Motivation** `api-helper` should always handle this error if everything is functioning correctly. This is meant to be less than Cloudflare to be able to show a Traefik-specific response.
```
Comment (MasterPtato) [inline docs/infrastructure/TIMEOUTS.md:33]: Is this meant to be the same text as the above block?

**Review by MasterPtato — CHANGES_REQUESTED** [review]

File: docs/infrastructure/TIMEOUTS.md (hunk 1 above)
Comment (NathanFlurry) [reply docs/infrastructure/TIMEOUTS.md]: Clarified the sections

**Review by MasterPtato — APPROVED** [review]

**Pull request merged**